src/app/tests/TestReadInteraction.cpp

```diff
@@ -72,6 +73,48 @@ void TestReadInteraction::GenerateReportData(nlTestSuite * apSuite, void * apCon
     err = reportDataBuilder.Init(&writer);
     NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
 
+    AttributeDataList::Builder attributeDataListBuilder = reportDataBuilder.CreateAttributeDataListBuilder();
```

Comment:
I don't understand why we we bothering making changes to this function; is its output ever used for anything?

Reply:
yep, the report data would be processed inside ProcessReportData, which is tested , then report.CheckSchemaValidity and  ProcessAttributeDataList is exercised  inside ProcessReportData